File: CS50/CS50/week3/factorial.c
#include <cs50.h>
#include <stdio.h>

int factorial (int number);
int main (void)
{
    // ptomt for number
    int n=get_int("Type a number: ");
    printf ("%i\n", factorial(n));

}

int factorial (int number)
{
    if (number ==1)
    {
        return 1;
    }
    else
    {
        number*=factorial(number -1);
    }
    return number;
    /*
    int solution =number;
    for (int i=number-1; i>0; i-- )
    {
        solution=solution*i;
    }
    */
}

File: Python_coding/LinkedIn/Essential/tasks/factorial.py

def factorial(num):
    # Your code goes here.
    if type(num) != int or num < 0:
        return None
    
    if num ==0:
        return 1
    
    
    fac = 1
    while num>0:
        fac=fac*num

        print(f"factorias is: {fac}")
        print(num)
        num-=1
        print(num)
    return fac

#  another solution:
#       return num*factorial(num-1)


number = 5
result = factorial(number)
print(result)